### PR TITLE
ContainerBuilder loading kube config

### PR DESCRIPTION
**Pull request by gaoning777**
it enables the ai platform notebook as well as local notebook connections.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1795)

### CODE REVIEW

File: sdk/python/kfp/compiler/_k8s_helper.py [hunk 1]
```diff
@@ -31,13 +31,15 @@ def __init__(self):
...
       config.load_incluster_config()
       logging.info('Initialized with in-cluster config.')
-    
+    except:
+      logging.info('Cannot find in-cluster config, trying the local kubernetes config. ')
+      try:
+        config.load_kube_config()
+        logging.info('Found local kubernetes config. Initialized with kube_config.')
+      except:
+        raise RuntimeError('Forgot to run the gcloud command?')
```
Comment (hongye-sun) [inline sdk/python/kfp/compiler/_k8s_helper.py]: Maybe show this link to the user: https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl

File: sdk/python/kfp/compiler/_container_builder.py [hunk 2]
```diff
@@ -18,32 +18,57 @@
...
-SERVICEACCOUNT_NAMESPACE = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+SERVICEACCOUNT_NAMESPACE = '/var/run/secrets/kubernetes.io/serviceaccount/namespace'
+GCS_STAGING_BLOB_DEFAULT_PREFIX = 'kfp_container_build_staging'
 
 class ContainerBuilder(object):
   """
   ContainerBuilder helps build a container image
   """
-  def __init__(self, gcs_staging, namespace=None):
+  def __init__(self, gcs_staging=None, create_bucket_if_not_exist=True, namespace=None):
```
Comment (hongye-sun) [inline sdk/python/kfp/compiler/_container_builder.py]: nit: do you have a use case that user might set it to False?

File: sdk/python/kfp/compiler/_k8s_helper.py (hunk 1 above)
Comment (gaoning777) [reply sdk/python/kfp/compiler/_k8s_helper.py]: I like this. Will add it.

File: sdk/python/kfp/compiler/_container_builder.py (hunk 2 above)
Comment (gaoning777) [reply sdk/python/kfp/compiler/_container_builder.py]: I'll address this in: https://github.com/kubeflow/pipelines/pull/1776

Comment (gaoning777) [reply sdk/python/kfp/compiler/_container_builder.py]: This branch is created on top of that one.

Comment (gaoning777) [reply sdk/python/kfp/compiler/_container_builder.py]: Maybe not, I'll remove it.

**Pull request merged**